sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -90,6 +90,8 @@ abstract class Optimizer(sessionCatalog: SessionCatalog, conf: CatalystConf)
       CombineUnions,
       // Constant folding and strength reduction
       NullPropagation,
+      FoldablePropagation,
+      CleanupAliases,
```

Comment:
I think a simpler way is just calling `CleanupAliases.execute(result)` at end of `FoldablePropagation`, instead of putting it into optimizer.